jujugui/static/gui/src/app/components/icon-list/icon-list.js

```diff
@@ -0,0 +1,105 @@
+/* Copyright (C) 2018 Canonical Ltd. */
+'use strict';
+
+const PropTypes = require('prop-types');
+const React = require('react');
+
+/**
+  Display a list of icons for an entity.
+*/
+class IconList extends React.Component {
+  constructor() {
+    super();
+  }
+
+  /**
+    Generate the store state for the item.
+    @param {String} id The entity id.
+    @returns {Object} The store state.
+  */
+  _generateStoreState(id) {
+    // TODO frankban: it should be clear whether this id is legacy or not.
```

Comment:
This is a new component so why not making it part of the contract already and handling ambiguity on the caller, rather then adding something to my TODO list? ;-)

Reply:
Not long ago I expanded the url lib to add this: https://github.com/juju/juju-gui/blob/develop/jujugui/static/gui/src/app/jujulib/urls.js#L342